Source/STL/STLDocument.cs

```diff
@@ -131,7 +133,8 @@ public static bool IsText(Stream stream)
             //Read the header as ASCII.
             header = Encoding.ASCII.GetString(buffer);
 
-            return solid.Equals(header, StringComparison.InvariantCultureIgnoreCase);
+            return solid.Equals(header, StringComparison.InvariantCultureIgnoreCase) &&
+                   facet.Equals(header, StringComparison.InvariantCultureIgnoreCase);
```

Comment:
But the header will never be equal to both `"solid"` and `"facet"`. Also you're reading 1024 bytes, which will never equal `"solid"` or `"facet"` either. I think you'd need to add some test cases to cover your changes and make sure it's doing what you expect.